specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json

```diff
@@ -44,6 +43,15 @@
       ],
       "properties": {}
     },
+    "IotHubDeviceTelemetryEventData": {
```

Comment:
Can you tell me where this is used or referenced? I do not find anywhere this is used.

Reply:
That could be said for other definitions as well like IotHubDeviceCreatedEventData, IotHubDeviceCreatedEventData, etc. Is there something I'm missing ?

Reply:
Sure. The same question can be asked for the definition IotHubDeviceCreatedEventData. That is the reason I am asking. WHat is the point of these definitions? How exactly a SDK customer will use this definition (which will be a class in the generated SDK?)

Reply:
@Kishp01 @kalyanaj

Reply:
These definitions become part of the generated EventGrid data plane SDKs. For example, using Event Grid, a customer can create an "event subscription" subscribing to an IotDeviceCreated event and providing a destination such as a HTTPS endpoint (e.g. an Azure function). EventGrid will deliver such events to the customer's endpoint. The customer's endpoint can then consume the received event using this EventGrid data plane SDK. For a sample (that uses StorageBlobCreatedEventData), please see https://github.com/Azure-Samples/event-grid-dotnet-publish-consume-events/blob/master/EventGridConsumer/EventGridConsumer/Function1.cs.